experimental/attestation_common/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use openssl::{error::ErrorStack, nid::Nid, x509::X509Extension};
+use serde::{Deserialize, Serialize};
+use x509_parser::{
+    der_parser::{self, oid::Oid},
+};
+
+// Using [`Nid::NETSCAPE_COMMENT`] identifier since `rust-openssl` doesn't support custom
+// extensions yet.
+// https://github.com/sfackler/rust-openssl/issues/1411
+// https://www.alvestrand.no/objectid/2.16.840.1.113730.1.13.html
+pub const TEE_EXTENSION_OID: Oid<'static> = der_parser::oid!(2.16.840.1.113730.1.13);
+
+/// Placeholder implementation of TEE quote for remote attestation.
+#[derive(Deserialize, Serialize, Debug)]
+pub struct Quote {
+    pub measurement: String,
+    pub public_key_hash: String,
+}
+
+impl Quote {
+    pub fn new(measurement: &str, public_key: &[u8]) -> Self {
+        Self {
+            measurement: measurement.to_string(),
+            public_key_hash: get_sha256_hex(public_key),
+        }
+    }
+
+    pub fn to_string(&self) -> String {
+        serde_json::to_string(&self).expect("Couldn't serialize TEE quote")
+    }
+
+    /// Return the TEE [`Quote`] as a custom [`X509Extension`].
+    pub fn to_extension(&self) -> Result<X509Extension, ErrorStack> {
+        // Using [`Nid::NETSCAPE_COMMENT`] identifier since `rust-openssl` doesn't support custom
+        // extensions yet:
+        // https://github.com/sfackler/rust-openssl/issues/1411
+        X509Extension::new_nid(None, None, Nid::NETSCAPE_COMMENT, &self.to_string())
+    }
+}
+
+impl std::str::FromStr for Quote {
+    type Err = serde_json::Error;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
```

Comment:
It may be better to just have this as a method on `Quote` itself, like `to_string` is already there. Or alternatively implement `Display` for it, or something like that.